src/directory-sync/directory-sync.ts

```diff
@@ -0,0 +1,54 @@
+import WorkOS from '../workos';
+import { Directory } from './interfaces/directory.interface';
+import { List } from '../common/interfaces/list.interface';
+import { Group } from './interfaces/group.interface';
+import { User } from './interfaces/user.interface';
+import { ListDirectoriesOptions } from './interfaces/list-directories-options.interface';
+import { PaginationOptions } from '../common/interfaces/pagination-options.interface';
+
+export class DirectorySync {
+  constructor(private readonly workos: WorkOS) {}
+
+  async listDirectories(
+    options?: ListDirectoriesOptions,
+  ): Promise<List<Directory>> {
+    const { data } = await this.workos.get('/directories', options);
+    return data;
+  }
+
+  async listGroups(
+    directoryID: string,
+    options?: PaginationOptions,
+  ): Promise<List<Group>> {
+    const { data } = await this.workos.get(
+      `/directories/${directoryID}/groups`,
+      options,
+    );
+    return data;
+  }
+
+  async listUsers(
+    directoryID: string,
+    options?: PaginationOptions,
+  ): Promise<List<User>> {
+    const { data } = await this.workos.get(
+      `/directories/${directoryID}/users`,
+      options,
+    );
+    return data;
+  }
+
+  async listUsersGroups(directoryID: string, userID: string): Promise<Group[]> {
```

Comment:
Should this be `listUserGroups`?

We're grabbing groups for a single user, right?

Reply:
Shouldn't the noun be possessive?

Reply:
I always think about it as singular/plural. Don't think I've ever seen a possessive in API naming.

Reply:
👍 https://github.com/workos-inc/workos-node/pull/126